Tools/ci-run.sh

```diff
@@ -3,125 +3,164 @@
 GCC_VERSION=${GCC_VERSION:=8}
 
 # Set up compilers
-if [ "${OS_NAME##ubuntu*}" == "" -a "$TEST_CODE_STYLE" != "1" ]; then
+if [[ $TEST_CODE_STYLE == "1" ]]; then
+  echo "Skipping compiler setup"
+elif [[ $OSTYPE == "linux-gnu"* ]]; then
+  echo "Setting up linux compiler"
   echo "Installing requirements [apt]"
   sudo apt-add-repository -y "ppa:ubuntu-toolchain-r/test"
   sudo apt update -y -q
   sudo apt install -y -q ccache gdb python-dbg python3-dbg gcc-$GCC_VERSION || exit 1
-  if [ -z "${BACKEND##*cpp*}" ]; then
+  if [[ $BACKEND == *"cpp"* ]]; then
     sudo apt install -y -q g++-$GCC_VERSION || exit 1
+    ALTERNATIVE_ARGS=" --slave /usr/bin/g++ g++ /usr/bin/g++-$GCC_VERSION"
   fi
   sudo /usr/sbin/update-ccache-symlinks
   echo "/usr/lib/ccache" >> $GITHUB_PATH # export ccache to path
 
-  sudo update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-$GCC_VERSION 60 $(if [ -z "${BACKEND##*cpp*}" ]; then echo " --slave /usr/bin/g++ g++ /usr/bin/g++-$GCC_VERSION"; fi)
+  sudo update-alternatives --install /usr/bin/gcc gcc /usr/bin/gcc-$GCC_VERSION 60 $ALTERNATIVE_ARGS
 
   export CC="gcc"
-  if [ -z "${BACKEND##*cpp*}" ]; then
+  if [[ $BACKEND == *"cpp"* ]]; then
     sudo update-alternatives --set g++ /usr/bin/g++-$GCC_VERSION
     export CXX="g++"
   fi
-fi
-if [ "${OS_NAME##macos*}" == "" ]; then
+elif [[ $OSTYPE == "darwin"* ]]; then
+  echo "Setting up macos compiler"
   export CC="clang -Wno-deprecated-declarations"
   export CXX="clang++ -stdlib=libc++ -Wno-deprecated-declarations"
 fi
 
 # Set up miniconda
-if [ "$STACKLESS" == "true" ]; then
+if [[ $STACKLESS == "true" ]]; then
   echo "Installing stackless python"
   #conda install --quiet --yes nomkl --file=test-requirements.txt --file=test-requirements-cpython.txt
   conda config --add channels stackless
   conda install --quiet --yes stackless || exit 1
 fi
 
+PYTHON_SYS_VERSION=$(python -c 'import sys; print(sys.version)')
+
 # Log versions in use
 echo "===================="
 echo "|VERSIONS INSTALLED|"
 echo "===================="
-python -c 'import sys; print("Python %s" % (sys.version,))'
-if [ "$CC" ]; then
-  which ${CC%% *}
-  ${CC%% *} --version
-fi
-if [ "$CXX" ]; then
+echo "Python $PYTHON_SYS_VERSION"
+if [[ $CXX ]]; then
   which ${CXX%% *}
   ${CXX%% *} --version
 fi
+if [[ $CC ]]; then
+  which ${CC%% *}
+  ${CC%% *} --version
+fi
```

Comment:
Why reverse the order here?

Reply:
Just .. well, in this context no reason, but the initial intention wat to use `if [[ $BACKEND == *"cpp"* ]]; then` but then i understood that such condition will not work as intended

Reply:
I'll reverse it back

Reply:
[resolved](https://github.com/cython/cython/pull/4324/commits/987395a98c165bdf251e3a1f3fd3c362acd82c19)